packages/lit-html/src/lit-html.ts

```diff
@@ -708,6 +706,41 @@ class Template {
   }
 }
 
+function resolveDirective(part: NodePart, value: unknown): unknown;
+function resolveDirective(
+  part: AttributePart,
+  value: unknown,
+  i: number
+): unknown;
+function resolveDirective(
+  part: NodePart | AttributePart,
+  value: unknown,
+  i?: number
+) {
+  let directive =
+    i != null
+      ? (part as AttributePart)._directives?.[i]
+      : (part as NodePart)._directive;
+  const directiveClass = isPrimitive(value)
+    ? undefined
+    : (value as DirectiveResult)?._$litDirective$;
+  if (directive?.constructor !== directiveClass) {
```

Comment:
`directive` is the current one but `directiveClass` is the new one and that's confusing. Maybe rename `directive` to `currentDirective` and `directiveClass` to `nextDirectiveConstructor`?

Reply:
Done